### ISSUE TITLE
K-9 stopped working on Android 10

### ISSUE THREAD

**Issue by gmanic** [k9mail/k-9#4186]
After Upgrade of Android to v 10 k-9 stopped working. 

### Expected behavior
K-9 starts up

### Actual behavior
Does not Start. 

### Steps to reproduce
1. Upgrade from Android 9 (latest Update) to 10
2. Tap k-9 icon
3.

### Environment
K-9 Mail version:
Latest
Android version:
10
Account type (IMAP, POP3, WebDAV/Exchange):
Imap
Please take some time to [retrieve logs](https://github.com/k9mail/k-9/wiki/LoggingErrors)
Does not Start...

**Comment by njeyaakili** [comment]
this has been reported previously with the 10/Q beta releases -- e.g., 4042. 

uninstalling and reinstalling, though not the most elegant approach, seems to resolve the problem.

**Comment by yeezy69** [comment]
SSLv3 problem on essential phone PH-1 after update to Android 10. The app can't connect to IMAP and SMTP server. Reinstalling didn't help.

On Mailserver there are following lines in logfile:

Sep  5 14:24:03 mailsrv postfix/smtpd[15035]: warning: TLS library problem:
error:14094417:SSL routines:ssl3_read_bytes:sslv3 alert illegal
parameter:../ssl/record/rec_layer_s3.c:1528:SSL alert number 47:

Sep  5 14:24:26 mailsrv dovecot: imap-login: Disconnected (no auth attempts
in 0 secs): user=<>, rip=x.x.x.x, lip=x.x.x.x, TLS
handshaking: SSL_accept() failed: error:14094417:SSL
routines:ssl3_read_bytes:sslv3 alert illegal parameter: SSL alert number
47, session=<LOkOZ82RI1JtKED3>

**Comment by njeyaakili** [comment]
the issue of k-9 not starting under android-10 is different than what you are reporting with the logs, above. i believe that what your logs are showing is a server-side issue related to openssl. i haven't followed this enough to be able to give advice, so you may want to ask on the postfix or openssl lists.  [as an aside, i don't have a problem sending/receiving mail with k-9 on my essential ph-1/android-10 device.]

**Comment by gmanic** [comment]
Unbelieveable. That makes us three ph-1 users with k9. 
My k-9 is ok again after reinstalling.

As a Mailserver side note: I had to realize that my unsuccessful connection attempts triggered fail2ban and banned my vpn endpoint IP for quite some time....
Strict vpn usage (No connect without vpn) triggers connect provs with g play eg...

**Comment by cketti** [comment]
Duplicate of #4042

**Comment by fizbin** [comment]
@yeezy69 I just solved an identical problem in my setup. In my case it was a subtle error in my server's SSL cert that most clients ignore, but the version of the SSL library shipped with Android 10 enforces not connecting to servers with this problem.

Specifically, look at your server's certificate for the value of the "X509v3 Key Usage" extension. To be used as a server SSL cert, "Key Usage" must have both "Digital Signature" and "Key Encipherment" (different SSL ciphers require different key usage from the server's cert). I was able to keep my server's private key, but had to generate a new CSR and then had to sign it again, adding the correct key usage this time.
